packages/php-wasm/compile/build-assets/php7.4.patch

```diff
@@ -34,5 +34,34 @@ diff --git a/php-src/ext/standard/file.c b/php-src/ext/standard/file.c
  		php_error_docref(NULL, E_WARNING, "The first argument to copy() function cannot be a directory");
  		return FAILURE;
  	}
- 
+
  	switch (php_stream_stat_path_ex(dest, PHP_STREAM_URL_STAT_QUIET | PHP_STREAM_URL_STAT_NOCACHE, &dest_s, ctx)) {
+
+diff --git a/php-src/main/streams/cast.c b/main/streams/cast.c
+index 3bad65fbac..05cab34658 100644
+--- a/php-src/main/streams/cast.c
++++ b/php-src/main/streams/cast.c
+@@ -102,8 +102,7 @@ static ssize_t stream_cookie_writer(void *cookie, const char *buffer, size_t siz
+ 	return php_stream_write(((php_stream *)cookie), (char *)buffer, size);
+ }
+
+-# ifdef COOKIE_SEEKER_USES_OFF64_T
+-static int stream_cookie_seeker(void *cookie, off64_t *position, int whence)
++static int stream_cookie_seeker(void *cookie, off_t *position, int whence)
```

Comment:
Nice! I know we need these patches to suppress emscripten errors, but I don't have a good understanding why. @seanmorris do you have any insights as to what might be root cause?

Reply:
@adamziel 
Actually it seems that under the hood, musl was using some macro-magic to map those to the standard names anyhow,  i.e. `off64_t` -> `off_t`. Musl is now deprecating those macros, so we need to account for that one way or another.

https://musl.libc.org/releases.html

> On the API level, the legacy "LFS64" ("large file support") interfaces, which were provided by macros remapping them to their standard names (#define stat64 stat and similar) have been deprecated and are no longer provided under the _GNU_SOURCE feature profile, only under explicit _LARGEFILE64_SOURCE. The latter will also be removed in a future version. Builds broken by this change can be fixed short-term by adding -D_LARGEFILE64_SOURCE to CFLAGS, but should be fixed to use the standard interfaces.

_**See also**_: https://github.com/php/php-src/issues/11678

Reply:
It sounds like we could provide our own mapping for these standard names instead of patching the code to use different names?

Reply:
@adamziel Yea that's completely possible. Am I correct in assuming you're suggesting we need to create macros to cover what we lost in the deprecation?

Reply:
@seanmorris yes! Perhaps even more than that. In my ideal world, we wouldn't need patches such as:

```
-static int pdo_sqlite_stmt_get_col(pdo_stmt_t *stmt, int colno, char **ptr, zend_ulong *len, int *caller_frees)
+static int pdo_sqlite_stmt_get_col(pdo_stmt_t *stmt, int colno, char **ptr, size_t *len, int *caller_frees)
```

That change `zend_ulong` -> `size_t`. Now, this one sounds like a separate problem whatsoever and may be entirely out of scope here, but perhaps it would just take one more macro to deal with that one as well?

Reply:
@adamziel I'd need to look into the reasoning behind that change before I can comment on it, but lets go ahead and open a github issue to track this, it doesn't seem like a big lift to alias a couple of names if that's all that needs to be done.

Reply:
@seanmorris I opened a separate issue to track this: https://github.com/WordPress/wordpress-playground/issues/652